.ci/test.sh

```diff
@@ -61,17 +62,15 @@ if [[ $TASK == "check-docs" ]] || [[ $TASK == "check-links" ]]; then
 fi
 
 if [[ $TASK == "lint" ]]; then
-    conda install -q -y -n $CONDA_ENV \
+    mamba install -q -y -n $CONDA_ENV \
+        cmakelint \
+        cpplint \
+        isort \
+        libxml2 \
```

Comment:
Could you please check whether we still need explicitly install `libxml2`?
https://github.com/microsoft/LightGBM/pull/2954#issuecomment-605233486

Reply:
oh yeah, nice! I'll be very happy if this PR is able to remove this and other installation hacks 🙌 

tried removing it in https://github.com/microsoft/LightGBM/pull/4953/commits/bc750fbd3523bd6250f35fd4538c0a94c8c03f4b, let's see how it goes.

Reply:
all linting tasks passed! https://github.com/microsoft/LightGBM/runs/4909444083?check_suite_focus=true

And I don't see any issues in the logs. So looks like it's ok to remove this `libxml2` install.